### PR TITLE
Refactor musical chairs to include random seat reduction

### PR DESCRIPTION
**Pull request by soaresa**
- Add SEATS_THRESHOLD and MAX_RANDOM_REDUCTION constants
- Introduce apply_maybe_random_reduction function
- Apply random seat reduction if seats exceed threshold
- Add unit tests for apply_maybe_random_reduction with various seat counts
- Ensure random reduction logic is well-tested with a fixed seed

### CODE REVIEW

File: framework/libra-framework/sources/ol_sources/musical_chairs.move [hunk 1]
```diff
@@ -177,6 +174,18 @@ module ol_framework::musical_chairs {
         (compliant_vals, seats_offered)
     }
 
+    // Applies a random reduction to the number of seats if the seats exceed the threshold.
+    // Reduces the seats by up to 10% to force competition among validators.
+    fun apply_maybe_random_reduction(seats: u64): u64 {
```
Comment (0o-de-lally) [inline framework/libra-framework/sources/ol_sources/musical_chairs.move:179]: One issue with this implementation is that it seems the validator seat selection will always shrink, when randomness is enabled. I think there would have to be a change elsewhere in how the max set size is calculated. Maybe there some state which remembers what was the last max size recommended.

Comment (soaresa) [reply framework/libra-framework/sources/ol_sources/musical_chairs.move:179]: I agree. My recent changes were mainly focused on refactoring to improve test coverage. However, if you are open to a new implementation, I would be happy to design a new model that conservatively adjusts the number of seats based on both current and past states.

Comment (soaresa) [reply framework/libra-framework/sources/ol_sources/musical_chairs.move:179]: Also, I believe we should have pure functions for the calculation of the number of seats; it is better for readability and testing. Pure functions also offer predictability, reusability, and simplified debugging.

Comment (0o-de-lally) [reply framework/libra-framework/sources/ol_sources/musical_chairs.move:179]: Yes there's a lot of spaghetti in the pof.move file. This is a good opportunity to clean. Also, yes, there's a seat offering adjustment which is possible relative to number of eligible bidders.

**Comment by 0o-de-lally** [comment]
I see this commit on community version, closing.

**Pull request closed without merging**